### PR TITLE
feat: dynamic btc fees

### PR DESCRIPTION
**Pull request by elmutt**
Dynamic btc fees from unchained

### CODE REVIEW

**Comment by cjthompson** [comment]
Does this PR require an unchained PR to be merged and deployed? There are tests failing which I assume is because an updated unchained-client is required.

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by elmutt** [comment]
> Does this PR require an unchained PR to be merged and deployed? There are tests failing which I assume is because an updated unchained-client is required.

File: packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.test.ts [hunk 1]
```diff
@@ -122,6 +118,23 @@ const getTransactionMockResponse = {
   }
 }
 
+const getNetworkFeesMockedResponse = {
+  data: {
+    fast: {
+      blocksUntilConfirmation: 1,
+      satsPerKiloByte: 1024
```
Comment (natven) [inline packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.test.ts:125]: what's up with sats per kilo? txs can be significantly less than 1k

Comment (elmutt) [reply packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.test.ts:125]: thats how it comes back from unchained

**Review by natven — APPROVED** [review]

File: packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.test.ts (hunk 1 above)
Comment (kaladinlight) [reply packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.test.ts:125]: Value being returned from estimatesmartfee call: https://bitcoincore.org/en/doc/0.21.0/rpc/util/estimatesmartfee/

File: packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts [hunk 2]
```diff
@@ -246,15 +246,29 @@ export class ChainAdapter implements IChainAdapter<ChainTypes.Bitcoin> {
...
+
+    if (
+      !feeData.data.fast?.satsPerKiloByte ||
+      !feeData.data.average?.satsPerKiloByte ||
+      !feeData.data.slow?.satsPerKiloByte
+    )
+      throw new Error('undefined fee')
+
+    // We have to round because coinselect library uses sats per byte which cant be decimals
+    const fast = String(Math.round(feeData.data.fast?.satsPerKiloByte / 1024))
```
Comment (pastaghost) [inline packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts:259]: Should we use Math.ceil() here instead to avoid underestimating fees?

**Pull request merged**